### DISCUSSION TITLE
information time of a picture

### DISCUSSION THREAD

**Discussion by SuperMotzer** [bpatrik/pigallery2#360] [Q&A]
Hello,
Browsing my pictures and showing the information, don't catch how the shown time is calculated. Never the original time of the pic.
Thank you.

**Comment by SuperMotzer** [comment]
Seems that the picture time is corrected with the time zone. All pictures are indicated 1 or 2 hours laterthan in reality.
Pictures coming from several devices (Eos, iPhone) show the same trouble.
Using latest version in Alpine form.

**Comment by bpatrik** [comment]
The app uses the creation data as it is in the exif/iptc data (or the file creation date if none of those exists). Eg.: https://github.com/bpatrik/pigallery2/blob/753cc9db774884b0c3cb8a4f704df9756aae1730/src/backend/model/threading/MetadataLoader.ts#L71

Using the local time, when it was shot.

**Comment by SuperMotzer** [comment]
Agree with you how it should work. But the results is not that. It seems to consider a timezone parameter
![Clipboard01](https://user-images.githubusercontent.com/90383260/134854068-4bd1e16d-81a6-47ba-adfe-dc5ee207e11f.png)
.

**Comment by SuperMotzer** [comment]
would appreciate some help for this topic. Mistake still available with latest-alpine on 11/24/21. Thank you.

**Comment by Sean-T-Moore** [comment]
I see this problem too using docker image: bpatrik/pigallery2:latest.
From the MetadataLoader code I see that pigallery2 correctly gets metadata.creationDate from my photos exif.tags.DateTimeOriginal. However, if my browser is running on a system with a timezone of anything other than GMT, the displayed photo time in pigallery2 Info will be incorrect by an amount equal to the TZ offset. For example, if my TZ offset is GMT-7, then the displayed time in pigallery2 is 7 hours before the DateTimeOriginal exif tag from the photo.  First I thought maybe this is due to the fact that docker container is always GMT, but mapping my /etc/timezone into docker container did not fix this problem. So perhaps something in the frontend code is performing a TZ offset of the creationDate?

**Comment by SuperMotzer** [comment]
Hello Sean,
Yes, thanks for the confirmation. I did same analysis 10 months ago using the current version from that time. Also changing the docker TZ settings did bring anything. As bpatrik didn't respond to my concrete example, I just move away from pigallery2.

**Reply by bpatrik** [reply]
just seeing this. 
I'm truly sorry that I cannot provide a better support for this app, but with work and life, I have less and less time for this project. I can spend a few hours here and there on this project lately.

**Comment by Sean-T-Moore** [comment]
Since this is apparent with a clean install, default settings, and using docker.... I would think this would affect everybody? It would be helpful to know if it doesn't affect some people so I would know if there is a setting or configuration that works.

**Comment by bpatrik** [comment]
related to #530

**Reply by bpatrik** [reply]
probably also related to #517

**Reply by bpatrik** [reply]
and pr #480 and issue #469

**Comment by bpatrik** [comment]
I do not  have time for the near future to fix this. You can try to use an earlier version if that solves the issue or I'm happily accept PRs

**Comment by Sean-T-Moore** [comment]
I see, PR #480 fixed this issue. It was merged into master on 2022-05-01 but docker image bpatrik/pigallery2:latest has not been updated since 2022-03-12. The latest tagged release 1.9.3 predates the fix. 
I was able to build and install from master today and verify this is fixed.
Thank you Patrik for the pointers and making this code available!

**Comment by bpatrik** [comment]
there were no recent release indeed, but the nightly builds: 
![kép](https://user-images.githubusercontent.com/7512182/187621193-f506e129-75a7-46a0-b845-4347cffbd273.png)

**Comment by SuperMotzer** [comment]
Thanks so much for the information. nightly versions solve the problem and I use Pigallery2 again.
